core/primitives/src/time.rs

```diff
@@ -179,9 +181,85 @@ impl Clock {
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub struct Time {
+    system_time: SystemTime,
+}
+
+impl BorshSerialize for Time {
```

Comment:
do we put Time somewhere in Borsh format already?

Reply:
Not in the `master` branch. In this `PR`, we start using it with `KnownPeerState`, which gets written to, and loaded from the database. We don't use it for network protoctol yet.

Reply:
right - so let's mention that Time's Borsch serialization is compatible with 'nanos' serialization - and let's add a test for that (serialize nanos, deserialize using borsch, compare)